tests/unit/test_charm.py

```diff
@@ -1,67 +1,100 @@
+        self.harness = Harness(MySQLRouterOperatorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
+        self.maxDiff = None
+        self.name = "mysqlrouter"
 
-    def test_config_changed(self):
-        self.assertEqual(list(self.harness.charm._stored.things), [])
-        self.harness.update_config({"thing": "foo"})
-        self.assertEqual(list(self.harness.charm._stored.things), ["foo"])
-
-    def test_action(self):
-        # the harness doesn't (yet!) help much with actions themselves
-        action_event = Mock(params={"fail": ""})
-        self.harness.charm._on_fortune_action(action_event)
-
-        self.assertTrue(action_event.set_results.called)
-
-    def test_action_fail(self):
-        action_event = Mock(params={"fail": "fail this"})
-        self.harness.charm._on_fortune_action(action_event)
-
-        self.assertEqual(action_event.fail.call_args, [("fail this",)])
-
-    def test_httpbin_pebble_ready(self):
-        # Check the initial Pebble plan is empty
-        initial_plan = self.harness.get_container_pebble_plan("httpbin")
-        self.assertEqual(initial_plan.to_yaml(), "{}\n")
-        # Expected plan after Pebble ready with default config
-        expected_plan = {
-            "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": "🎁"},
-                }
-            },
-        }
-        # Get the httpbin container from the model
-        container = self.harness.model.unit.get_container("httpbin")
-        # Emit the PebbleReadyEvent carrying the httpbin container
-        self.harness.charm.on.httpbin_pebble_ready.emit(container)
-        # Get the plan now we've run PebbleReady
-        updated_plan = self.harness.get_container_pebble_plan("httpbin").to_dict()
-        # Check we've got the plan we expected
-        self.assertEqual(expected_plan, updated_plan)
-        # Check the service was started
-        service = self.harness.model.unit.get_container("httpbin").get_service("httpbin")
-        self.assertTrue(service.is_running())
-        # Ensure we set an ActiveStatus with no message
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus())
+    @patch("charm.MySQLRouterOperatorCharm._install_apt_packages")
+    @patch("charm.MySQLRouterOperatorCharm._install_snap_packages")
+    def test_on_install(self, _install_snap_packages, _install_apt_packages):
+        """Test the on_install method."""
+        self.harness.charm.on.install.emit()
+        _install_apt_packages.assert_called_once()
+        _install_snap_packages.assert_called_once()
+
+        self.assertEqual(
+            self.harness.charm.unit.status, WaitingStatus("waiting for database relation")
```

Comment:
nit (this allows for more flexible testing, ie if someone changes the waiting message then the tests won't fail): 
```suggestion
            self.assertTrue(isinstance(self.harness.charm.unit.status, WaitingStatus))
```

Reply:
Done.